src/ecs/system_manager.cpp

```diff
@@ -0,0 +1,22 @@
+#include "ecs/system_manager.h"
+
+using namespace arch::ecs;
+
+void SystemManager::on_entity_destroyed(EntityId entity_id) {
+	for (auto const& system : _systems) {
+		system->entity_ids.erase(entity_id);
+	}
+}
+
+void SystemManager::on_entity_mask_change(EntityId entity_id,
+										  ComponentMask entity_mask) {

```

Comment:
wtf

Reply:
?

Reply:
EntityId entity_id, <a lot of whitespace, newline, a lot of whitespace, ComponentMask enitity_mask>. It barely fits into snippet, so you have probably overlooked it.

Reply:
Yeah it starts at exactly the same space on the line below so I don't see the issue